Update dependency on elastic-package

mtojek opened the pull request:
This PR updates dependency on elastic-package to include latest changes with multiple variants support.

elasticmachine commented:
## :broken_heart: Tests Failed


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-1319/detail/PR-1319/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-1319/detail/PR-1319/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-1319/detail/PR-1319/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-1319/detail/PR-1319/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://integrations_1319.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/beats-ci/transactions/view?rangeFrom=2021-07-12T07:54:02.805Z&rangeTo=2021-07-12T08:14:02.805Z&transactionName=Ingest-manager/integrations/PR-{number}&transactionType=unknown&latencyAggregationType=avg&traceId=5d158977f1000d027c83701dc1694472&transactionId=ff5f112b737a4cd5)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2021-07-12T08:04:02.805+0000


* Duration: 80 min 21 sec


* Commit: 23f067b373543e6bc98e5ed4e71ef72f459d1e3c



#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 4  |
| Passed       | 2360  |
| Skipped      | 3 |
| Total        | 2367   |


#### Trends :test_tube:


![Image of Build Times](https://beats-ci.elastic.co/job/Ingest-manager/job/integrations/job/PR-1319/buildTimeGraph/png)


![Image of Tests](https://beats-ci.elastic.co/job/Ingest-manager/job/integrations/job/PR-1319/1/testReport/history/countGraph/png?start=0&amp;end=25)


</p>
</details>

<!-- TEST RESULTS IF ANY-->

  ### Test errors [![4](https://img.shields.io/badge/4%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-1319/detail/PR-1319/1//tests)
<details><summary>Expand to view the tests failures</summary><p>
    
    
##### `Check integrations / mysql / mysql: check / system test: default (variant: percona_8_0_13) – mysql.error`


<ul>
<li>no error details</li>
<details><summary>Expand to view the stacktrace</summary><p>

```
 could not find hits in logs-mysql.error-ep data stream 
 ```
</p></details>
</ul>
    
##### `Check integrations / mysql / mysql: check / system test: default (variant: mariadb_10_2_23) – mysql.error`


<ul>
<li>no error details</li>
<details><summary>Expand to view the stacktrace</summary><p>

```
 could not find hits in logs-mysql.error-ep data stream 
 ```
</p></details>
</ul>
    
##### `Check integrations / mysql / mysql: check / system test: default (variant: mariadb_10_4_4) – mysql.error`


<ul>
<li>no error details</li>
<details><summary>Expand to view the stacktrace</summary><p>

```
 could not find hits in logs-mysql.error-ep data stream 
 ```
</p></details>
</ul>
    
##### `Check integrations / mysql / mysql: check / system test: default (variant: mysql_8_0_13) – mysql.error`


<ul>
<li>no error details</li>
<details><summary>Expand to view the stacktrace</summary><p>

```
 could not find hits in logs-mysql.error-ep data stream 
 ```
</p></details>
</ul>
    
</p></details>


<!-- STEPS ERRORS IF ANY -->









### Steps errors [![1](https://img.shields.io/badge/1%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-1319/detail/PR-1319/1//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


  
  
  
  
##### `Test integration: mysql`
<ul>
<li>Took 62 min 28 sec . View more details on <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Ingest-manager/pipelines/integrations/pipelines/PR-1319/runs/1/steps/4527/log/?start=0">here</a></li>
<li>Description: <code>                                eval "$(../../build/elastic-package stack shellinit)"                                ../../build/elastic-package test -v --report-format xUnit --report-output file                                </code></l1>
</ul>
  
  </p>
  </details>



<!-- LOG OUTPUT-->
### Log output

<details><summary>Expand to view the last 100 lines of log output</summary>
<p>

```
[2021-07-12T09:23:29.528Z] 2021/07/12 09:23:29 DEBUG adding package data stream to test policy...
[2021-07-12T09:23:29.528Z] 2021/07/12 09:23:29 DEBUG POST http://127.0.0.1:5601/api/fleet/package_policies
[2021-07-12T09:23:32.062Z] 2021/07/12 09:23:31 DEBUG deleting old data in data stream...
[2021-07-12T09:23:32.062Z] 2021/07/12 09:23:31 DEBUG found 0 hits in metrics-mysql.status-ep data stream
[2021-07-12T09:23:32.062Z] 2021/07/12 09:23:31 DEBUG GET http://127.0.0.1:5601/api/fleet/agent_policies/cf4c4fe0-e2f2-11eb-b1dd-d944493950f8
[2021-07-12T09:23:32.062Z] 2021/07/12 09:23:31 DEBUG assigning package data stream to agent...
[2021-07-12T09:23:32.062Z] 2021/07/12 09:23:31 DEBUG PUT http://127.0.0.1:5601/api/fleet/agents/c046baf4-fb88-45fc-a817-eee032a7eaee/reassign
[2021-07-12T09:23:32.630Z] 2021/07/12 09:23:32 DEBUG GET http://127.0.0.1:5601/api/fleet/agents/c046baf4-fb88-45fc-a817-eee032a7eaee
[2021-07-12T09:23:32.630Z] 2021/07/12 09:23:32 DEBUG Wait until the policy (ID: cf4c4fe0-e2f2-11eb-b1dd-d944493950f8, revision: 2) is assigned to the agent (ID: c046baf4-fb88-45fc-a817-eee032a7eaee)...
[2021-07-12T09:23:34.532Z] 2021/07/12 09:23:34 DEBUG GET http://127.0.0.1:5601/api/fleet/agents/c046baf4-fb88-45fc-a817-eee032a7eaee
[2021-07-12T09:23:34.532Z] 2021/07/12 09:23:34 DEBUG Wait until the policy (ID: cf4c4fe0-e2f2-11eb-b1dd-d944493950f8, revision: 2) is assigned to the agent (ID: c046baf4-fb88-45fc-a817-eee032a7eaee)...
[2021-07-12T09:23:37.075Z] 2021/07/12 09:23:36 DEBUG GET http://127.0.0.1:5601/api/fleet/agents/c046baf4-fb88-45fc-a817-eee032a7eaee
[2021-07-12T09:23:37.075Z] 2021/07/12 09:23:36 DEBUG Wait until the policy (ID: cf4c4fe0-e2f2-11eb-b1dd-d944493950f8, revision: 2) is assigned to the agent (ID: c046baf4-fb88-45fc-a817-eee032a7eaee)...
[2021-07-12T09:23:38.981Z] 2021/07/12 09:23:38 DEBUG GET http://127.0.0.1:5601/api/fleet/agents/c046baf4-fb88-45fc-a817-eee032a7eaee
[2021-07-12T09:23:38.981Z] 2021/07/12 09:23:38 DEBUG Wait until the policy (ID: cf4c4fe0-e2f2-11eb-b1dd-d944493950f8, revision: 2) is assigned to the agent (ID: c046baf4-fb88-45fc-a817-eee032a7eaee)...
[2021-07-12T09:23:40.884Z] 2021/07/12 09:23:40 DEBUG GET http://127.0.0.1:5601/api/fleet/agents/c046baf4-fb88-45fc-a817-eee032a7eaee
[2021-07-12T09:23:40.884Z] 2021/07/12 09:23:40 DEBUG Policy revision assigned to the agent (ID: c046baf4-fb88-45fc-a817-eee032a7eaee)...
[2021-07-12T09:23:40.884Z] 2021/07/12 09:23:40 DEBUG checking for expected data in data stream...
[2021-07-12T09:23:40.884Z] 2021/07/12 09:23:40 DEBUG found 1 hits in metrics-mysql.status-ep data stream
[2021-07-12T09:23:40.884Z] 2021/07/12 09:23:40 DEBUG reassigning original policy back to agent...
[2021-07-12T09:23:40.884Z] 2021/07/12 09:23:40 DEBUG PUT http://127.0.0.1:5601/api/fleet/agents/c046baf4-fb88-45fc-a817-eee032a7eaee/reassign
[2021-07-12T09:23:41.451Z] 2021/07/12 09:23:41 DEBUG GET http://127.0.0.1:5601/api/fleet/agents/c046baf4-fb88-45fc-a817-eee032a7eaee
[2021-07-12T09:23:41.451Z] 2021/07/12 09:23:41 DEBUG Wait until the policy (ID: ff20a940-e2e9-11eb-b1dd-d944493950f8, revision: 1) is assigned to the agent (ID: c046baf4-fb88-45fc-a817-eee032a7eaee)...
[2021-07-12T09:23:43.985Z] 2021/07/12 09:23:43 DEBUG GET http://127.0.0.1:5601/api/fleet/agents/c046baf4-fb88-45fc-a817-eee032a7eaee
[2021-07-12T09:23:43.986Z] 2021/07/12 09:23:43 DEBUG Wait until the policy (ID: ff20a940-e2e9-11eb-b1dd-d944493950f8, revision: 1) is assigned to the agent (ID: c046baf4-fb88-45fc-a817-eee032a7eaee)...
[2021-07-12T09:23:45.887Z] 2021/07/12 09:23:45 DEBUG GET http://127.0.0.1:5601/api/fleet/agents/c046baf4-fb88-45fc-a817-eee032a7eaee
[2021-07-12T09:23:45.887Z] 2021/07/12 09:23:45 DEBUG Wait until the policy (ID: ff20a940-e2e9-11eb-b1dd-d944493950f8, revision: 1) is assigned to the agent (ID: c046baf4-fb88-45fc-a817-eee032a7eaee)...
[2021-07-12T09:23:47.791Z] 2021/07/12 09:23:47 DEBUG GET http://127.0.0.1:5601/api/fleet/agents/c046baf4-fb88-45fc-a817-eee032a7eaee
[2021-07-12T09:23:47.791Z] 2021/07/12 09:23:47 DEBUG Wait until the policy (ID: ff20a940-e2e9-11eb-b1dd-d944493950f8, revision: 1) is assigned to the agent (ID: c046baf4-fb88-45fc-a817-eee032a7eaee)...
[2021-07-12T09:23:49.694Z] 2021/07/12 09:23:49 DEBUG GET http://127.0.0.1:5601/api/fleet/agents/c046baf4-fb88-45fc-a817-eee032a7eaee
[2021-07-12T09:23:49.694Z] 2021/07/12 09:23:49 DEBUG Policy revision assigned to the agent (ID: c046baf4-fb88-45fc-a817-eee032a7eaee)...
[2021-07-12T09:23:49.694Z] 2021/07/12 09:23:49 DEBUG deleting test policy...
[2021-07-12T09:23:49.694Z] 2021/07/12 09:23:49 DEBUG POST http://127.0.0.1:5601/api/fleet/agent_policies/delete
[2021-07-12T09:23:52.998Z] 2021/07/12 09:23:52 DEBUG tearing down service...
[2021-07-12T09:23:52.998Z] 2021/07/12 09:23:52 DEBUG tearing down service using docker compose runner
[2021-07-12T09:23:52.998Z] 2021/07/12 09:23:52 DEBUG running command: /usr/local/bin/docker-compose -f /var/lib/jenkins/workspace/est-manager_integrations_PR-1319/src/github.com/elastic/integrations/packages/mysql/_dev/deploy/docker/docker-compose.yml -p elastic-package-service down --volumes
[2021-07-12T09:23:53.256Z] Stopping elastic-package-service_mysql_1 ... 
[2021-07-12T09:23:56.547Z] 
Stopping elastic-package-service_mysql_1 ... done
Removing elastic-package-service_mysql_is_ready_1 ... 
[2021-07-12T09:23:56.547Z] Removing elastic-package-service_mysql_1          ... 
[2021-07-12T09:23:56.547Z] 
Removing elastic-package-service_mysql_1          ... done

Removing elastic-package-service_mysql_is_ready_1 ... done
Removing network elastic-package-service_default
[2021-07-12T09:23:56.547Z] Removing volume elastic-package-service_mysqldata
[2021-07-12T09:23:56.547Z] 2021/07/12 09:23:56 DEBUG deleting data in data stream...
[2021-07-12T09:23:56.547Z] Error: one or more test cases failed
[2021-07-12T09:23:56.755Z] Running in /var/lib/jenkins/workspace/est-manager_integrations_PR-1319/src/github.com/elastic/integrations
[2021-07-12T09:23:56.795Z] Archiving artifacts
[2021-07-12T09:23:57.022Z] Recording test results
[2021-07-12T09:23:57.363Z] [Checks API] No suitable checks publisher found.
[2021-07-12T09:23:57.668Z] + build/elastic-package stack dump -v --output build/elastic-stack-dump/latest/mysql
[2021-07-12T09:23:57.668Z] 2021/07/12 09:23:57  WARN CommitHash is undefined, in both /var/lib/jenkins/workspace/est-manager_integrations_PR-1319/.elastic-package/version and the compiled binary, config may be out of date.
[2021-07-12T09:23:57.668Z] 2021/07/12 09:23:57 DEBUG Enable verbose logging
[2021-07-12T09:23:57.668Z] 2021/07/12 09:23:57 DEBUG Dump Elastic stack data
[2021-07-12T09:23:57.668Z] 2021/07/12 09:23:57 DEBUG Dump stack logs
[2021-07-12T09:23:57.668Z] 2021/07/12 09:23:57 DEBUG Recreate the output location (path: build/elastic-stack-dump/latest/mysql)
[2021-07-12T09:23:57.668Z] 2021/07/12 09:23:57 DEBUG Dump stack logs for elasticsearch
[2021-07-12T09:23:57.668Z] 2021/07/12 09:23:57 DEBUG running command: /usr/local/bin/docker-compose -f /var/lib/jenkins/workspace/est-manager_integrations_PR-1319/.elastic-package/profiles/default/stack/snapshot.yml -p elastic-package-stack logs elasticsearch
[2021-07-12T09:23:58.604Z] 2021/07/12 09:23:58 DEBUG Dump stack logs for elastic-agent
[2021-07-12T09:23:58.605Z] 2021/07/12 09:23:58 DEBUG running command: /usr/local/bin/docker-compose -f /var/lib/jenkins/workspace/est-manager_integrations_PR-1319/.elastic-package/profiles/default/stack/snapshot.yml -p elastic-package-stack logs elastic-agent
[2021-07-12T09:23:59.543Z] 2021/07/12 09:23:59 DEBUG Dump stack logs for fleet-server
[2021-07-12T09:23:59.543Z] 2021/07/12 09:23:59 DEBUG running command: /usr/local/bin/docker-compose -f /var/lib/jenkins/workspace/est-manager_integrations_PR-1319/.elastic-package/profiles/default/stack/snapshot.yml -p elastic-package-stack logs fleet-server
[2021-07-12T09:24:00.479Z] 2021/07/12 09:24:00 DEBUG Dump stack logs for kibana
[2021-07-12T09:24:00.479Z] 2021/07/12 09:24:00 DEBUG running command: /usr/local/bin/docker-compose -f /var/lib/jenkins/workspace/est-manager_integrations_PR-1319/.elastic-package/profiles/default/stack/snapshot.yml -p elastic-package-stack logs kibana
[2021-07-12T09:24:01.856Z] 2021/07/12 09:24:01 DEBUG Dump stack logs for package-registry
[2021-07-12T09:24:01.856Z] 2021/07/12 09:24:01 DEBUG running command: /usr/local/bin/docker-compose -f /var/lib/jenkins/workspace/est-manager_integrations_PR-1319/.elastic-package/profiles/default/stack/snapshot.yml -p elastic-package-stack logs package-registry
[2021-07-12T09:24:03.243Z] Path to stack dump: build/elastic-stack-dump/latest/mysql
[2021-07-12T09:24:03.243Z] Done
[2021-07-12T09:24:03.275Z] Archiving artifacts
[2021-07-12T09:24:03.705Z] + build/elastic-package stack down -v
[2021-07-12T09:24:03.705Z] 2021/07/12 09:24:03  WARN CommitHash is undefined, in both /var/lib/jenkins/workspace/est-manager_integrations_PR-1319/.elastic-package/version and the compiled binary, config may be out of date.
[2021-07-12T09:24:03.705Z] 2021/07/12 09:24:03 DEBUG Enable verbose logging
[2021-07-12T09:24:03.705Z] Take down the Elastic stack
[2021-07-12T09:24:03.705Z] 2021/07/12 09:24:03 DEBUG running command: /usr/local/bin/docker-compose -f /var/lib/jenkins/workspace/est-manager_integrations_PR-1319/.elastic-package/profiles/default/stack/snapshot.yml -p elastic-package-stack down --volumes
[2021-07-12T09:24:04.273Z] The ELASTICSEARCH_IMAGE_REF variable is not set. Defaulting to a blank string.
[2021-07-12T09:24:04.273Z] The KIBANA_IMAGE_REF variable is not set. Defaulting to a blank string.
[2021-07-12T09:24:04.273Z] The STACK_PATH variable is not set. Defaulting to a blank string.
[2021-07-12T09:24:04.273Z] The PROFILE_NAME variable is not set. Defaulting to a blank string.
[2021-07-12T09:24:04.273Z] The ELASTIC_AGENT_IMAGE_REF variable is not set. Defaulting to a blank string.
[2021-07-12T09:24:04.273Z] Stopping elastic-package-stack_elastic-agent_1    ... 
[2021-07-12T09:24:04.273Z] Stopping elastic-package-stack_fleet-server_1     ... 
[2021-07-12T09:24:04.273Z] Stopping elastic-package-stack_kibana_1           ... 
[2021-07-12T09:24:04.273Z] Stopping elastic-package-stack_package-registry_1 ... 
[2021-07-12T09:24:04.273Z] Stopping elastic-package-stack_elasticsearch_1    ... 
[2021-07-12T09:24:21.705Z] 
Stopping elastic-package-stack_elastic-agent_1    ... done

Stopping elastic-package-stack_fleet-server_1     ... done

Stopping elastic-package-stack_kibana_1           ... done

Stopping elastic-package-stack_package-registry_1 ... done

Stopping elastic-package-stack_elasticsearch_1    ... done
Removing elastic-package-stack_elastic-agent_is_ready_1    ... 
[2021-07-12T09:24:21.705Z] Removing elastic-package-stack_elastic-agent_1             ... 
[2021-07-12T09:24:21.705Z] Removing elastic-package-stack_fleet-server_is_ready_1     ... 
[2021-07-12T09:24:21.705Z] Removing elastic-package-stack_fleet-server_1              ... 
[2021-07-12T09:24:21.705Z] Removing elastic-package-stack_kibana_is_ready_1           ... 
[2021-07-12T09:24:21.705Z] Removing elastic-package-stack_kibana_1                    ... 
[2021-07-12T09:24:21.705Z] Removing elastic-package-stack_elasticsearch_is_ready_1    ... 
[2021-07-12T09:24:21.705Z] Removing elastic-package-stack_package-registry_is_ready_1 ... 
[2021-07-12T09:24:21.705Z] Removing elastic-package-stack_package-registry_1          ... 
[2021-07-12T09:24:21.705Z] Removing elastic-package-stack_elasticsearch_1             ... 
[2021-07-12T09:24:21.705Z] 
Removing elastic-package-stack_elastic-agent_is_ready_1    ... done

Removing elastic-package-stack_elasticsearch_is_ready_1    ... done

Removing elastic-package-stack_package-registry_is_ready_1 ... done

Removing elastic-package-stack_package-registry_1          ... done

Removing elastic-package-stack_fleet-server_is_ready_1     ... done

Removing elastic-package-stack_kibana_is_ready_1           ... done

Removing elastic-package-stack_kibana_1                    ... done

Removing elastic-package-stack_elasticsearch_1             ... done

Removing elastic-package-stack_elastic-agent_1             ... done

Removing elastic-package-stack_fleet-server_1              ... done
Removing network elastic-package-stack_default
[2021-07-12T09:24:21.705Z] Done
[2021-07-12T09:24:21.846Z] Failed in branch mysql
[2021-07-12T09:24:23.207Z] Running on Jenkins in /var/lib/jenkins/workspace/est-manager_integrations_PR-1319
[2021-07-12T09:24:23.287Z] [INFO] getVaultSecret: Getting secrets
[2021-07-12T09:24:23.334Z] Masking supported pattern matches of $VAULT_ADDR or $VAULT_ROLE_ID or $VAULT_SECRET_ID
[2021-07-12T09:24:24.210Z] + chmod 755 generate-build-data.sh
[2021-07-12T09:24:24.211Z] + ./generate-build-data.sh https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Ingest-manager/integrations/PR-1319/ https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Ingest-manager/integrations/PR-1319/runs/1 FAILURE 4821126
[2021-07-12T09:24:24.211Z] INFO: curl https://beats-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/Ingest-manager/integrations/PR-1319/runs/1/steps/?limit=10000 -o steps-info.json

```

</p>
</details>



<!--COMMENT_GENERATED_WITH_ID_comment.id-->

mtojek commented:
Ok, it seems that updating service variants went well for most of integrations, something is wrong with mysql integrations (FYI @jsoriano). I will investigate what's wrong with it.

mtojek commented:
/test

mtojek commented:
Ok, I updated the Docker image for MySQL to dynamically create `.cnf` file for error logs. I've changed also healthcheck to collect more error entries in the runtime:

```
2021-07-12T12:45:37.400625Z 40 [Note] Access denied for user 'root'@'localhost' (using password: YES)
2021-07-12T12:45:38.481326Z 42 [Note] Access denied for user 'root'@'localhost' (using password: YES)
2021-07-12T12:45:39.570612Z 44 [Note] Access denied for user 'root'@'localhost' (using password: YES)
2021-07-12T12:45:40.656779Z 46 [Note] Access denied for user 'root'@'localhost' (using password: YES)
2021-07-12T12:45:41.746385Z 48 [Note] Access denied for user 'root'@'localhost' (using password: YES)
2021-07-12T12:45:42.826265Z 50 [Note] Access denied for user 'root'@'localhost' (using password: YES)
2021-07-12T12:45:43.905627Z 52 [Note] Access denied for user 'root'@'localhost' (using password: YES)
```

jsoriano reviewed (approved):
Nice! :+1:

mtojek commented on packages/mysql/_dev/deploy/docker/healthcheck.sh:
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# Simulate hacking attempt to put some entries in the error log
+mysql -u root -pbad_password -hlocalhost -P 3306 > /dev/null
```
Frankly speaking this one truly solved the problem related to https://github.com/elastic/elastic-package/issues/405 . Without this change I was able to run successfully all system tests locally, but failed all CI runs.
